### PR TITLE
feat: add enhanced `localStorage` support check

### PR DESCRIPTION
**Pull request by hf**
Based on feedback from #436.

### CODE REVIEW

File: src/lib/helpers.ts [hunk 1]
```diff
@@ -15,6 +15,51 @@ export function uuid() {
...
+  try {
+    if (typeof globalThis.localStorage !== 'object') {
+      return false
+    }
+  } catch (e) {
+    // DOM exception when accessing `localStorage`
+    return false
+  }
+
+  if (localStorageWriteTests.tested) {
```
Comment (J0) [inline src/lib/helpers.ts:40]: When will this be true?

**Review by J0 — APPROVED** [review]
LGTM, thanks!

**Comment by tjessessky** [comment]
Sorry, I'm going to answer my own comment in case anyone finds themselves with the same problem I had. I wrote a gist here with sample code, but the basic fix is that you can pass a custom Storage class object into the createClient options, and use whatever storage methods you'd like. [See the code here](https://gist.github.com/tjessessky/87ad344ae39206a3063425c79294da6d).

File: src/lib/helpers.ts (hunk 1 above)
Comment (hf) [reply src/lib/helpers.ts:40]: It's set right below when the write test finishes.

**Pull request merged**